docs/compilers/CSharp/Compiler Breaking Changes - DotNet 10.md

```diff
@@ -156,3 +156,30 @@ struct S
     public static void M([UnscopedRef] ref int x) { }
 }
 ```
+
+## `record` and `record struct` types cannot define pointer type members, even when providing their own Equals implementations
+
+PROTOTYPE: Figure what version this is going into
+
+***Introduced in Visual Studio 2022 version 17.14***
+
+The specification for `record class` and `record struct` types indicated that any unsafe types are disallowed. However, this was not
+enforced correctly in 2 scenarios:
+
+1. When the `record class` or `record struct` type defined its own `Equals` implementation.
+2. When the field type only used the unsafe type in a nested context, such as `int*[]`.
```

Comment:
>When the field type only used the unsafe type in a nested context, such as `int*[]`. [](http://example.com/codeflow?start=3&length=84)

I couldn't find clear definition of "unsafe type" in the language spec. Therefore, it is not clear whether such array type should be considered as unsafe.